pkg/api/networkservice/mechanisms/kernel/helpers.go

```diff
@@ -70,38 +40,49 @@ func New(netnsURL string) *networkservice.Mechanism {
 	}
 }
 
-func (m *mechanism) GetParameters() map[string]string {
-	if m == nil {
-		return nil
+// ToMechanism converts unified mechanism to helper
+func ToMechanism(m *networkservice.Mechanism) *Mechanism {
+	if m.GetType() == MECHANISM {
+		return &Mechanism{
+			m,
+		}
+	}
+	return nil
+}
+
+// GetParameters returns the map of all parameters to the mechanism
+func (m *Mechanism) GetParameters() map[string]string {
+	if m.Parameters == nil {
+		m.Parameters = map[string]string{}
 	}
 	return m.Parameters
 }
 
-func (m *mechanism) GetNetNSInode() string {
-	if m == nil || m.GetParameters() == nil {
-		return ""
-	}
-	return m.GetParameters()[common.NetNSInodeKey]
+// GetNetNSInode returns the NetNS inode
+func (m *Mechanism) GetNetNSInode() string {
+	return m.GetParameters()[NetNSInodeKey]
 }
 
-// GetPCIAddress returns PCI address of the device
-func (m *mechanism) GetPCIAddress() string {
-	if m == nil || m.GetParameters() == nil {
-		return ""
-	}
-	return m.GetParameters()[PCIAddress]
+// SetNetNSInode sets the NetNS inode
+func (m *Mechanism) SetNetNSInode(netNSInode string) {
+	m.GetParameters()[NetNSInodeKey] = netNSInode
 }
 
-// IsPCIDevice - true if this mechanism is for a PCI device
-func (m *mechanism) IsPCIDevice() bool {
```

Comment:
Curious... why is IsPCIDevice() gone?

Reply:
I guess it has been added for future SR-IOV, but here we are working on SR-IOV and don't really need such thing...
OK, there is no need to remove it, we only need to change `== ""` to `!= ""`.